dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -922,17 +922,17 @@ private static ChatRequestMessage GetRequestMessage(ChatMessageContent message)
                 return new ChatRequestToolMessage(message.Content, toolIdString);
             }
 
-            if (message.Items is { Count: > 0 })
+            if (message.Items is { Count: 1 } && message.Items.FirstOrDefault() is TextContent textContent)
             {
-                return new ChatRequestUserMessage(message.Items.Select(static (KernelContent item) => (ChatMessageContentItem)(item switch
-                {
-                    TextContent textContent => new ChatMessageTextContentItem(textContent.Text),
-                    ImageContent imageContent => new ChatMessageImageContentItem(imageContent.Uri),
-                    _ => throw new NotSupportedException($"Unsupported chat message content type '{item.GetType()}'.")
-                })));
+                return new ChatRequestUserMessage(textContent.Text);
             }
 
-            return new ChatRequestUserMessage(message.Content);
+            return new ChatRequestUserMessage(message.Items.Select(static (KernelContent item) => (ChatMessageContentItem)(item switch
```

Comment:
We need to ensure that the Items property is not null, since it can be assigned a null value via public setter up the stack and a null reference exception will be thrown here.

Reply:
https://github.com/microsoft/semantic-kernel/blob/21ce67608cd666ef630f81cc9fc91777471025a0/dotnet/src/SemanticKernel.Abstractions/Contents/ChatMessageContent.cs#L64-L68

The code here shows `Items` can never be null. Even if they are manually set to null via setter, they are reset to an collection in the getter.

Reply:
Yes, right, resolving the comment.